.github/workflows/release.yml

```diff
@@ -1,32 +1,56 @@
-name: Post Publish
+name: Release
+
 on:
-  release:
-    types: [published]
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+
+permissions:
+  contents: write
+
 jobs:
-  tidy:
-    name: Tidy Asana
-    runs-on: ubuntu-latest
+  go-version:
+    runs-on: macos-latest
+    outputs:
+      version: ${{ steps.go-version.outputs.version }}
     steps:
-      - uses: breathingdust/github-asana-tidy@v1
-        with:
-          asana_pat: ${{ secrets.asana_pat }}
-          asana_target_section_gid: '1141945723817371'
-          asana_workspace_gid: '90955849329269'
-          asana_project_gid: '632425409545160'
-          asana_github_url_field_gid: '1134594824474912'
-          github_release_name: ${{ github.event.release.tag_name }}
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-  project-archive:
-    name: Archive Released Cards
-    runs-on: ubuntu-latest
+      - uses: actions/checkout@v2
+      - id: go-version
+        run: echo "::set-output name=version::$(cat ./.go-version)"
+  release-notes:
+    runs-on: macos-latest
     steps:
-      - uses: breathingdust/github-project-archive@v1
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Generate Release Notes
+        run: sed -n -e "1{/# /d;}" -e "2{/^$/d;}" -e "/# \[$(git describe --abbrev=0 --exclude="$(git describe --abbrev=0 --match='v*.*.*' --tags)" --match='v*.*.*' --tags | tr -d v)/q;p" CHANGELOG.md > release-notes.txt
+      - uses: actions/upload-artifact@v2
         with:
-          github_done_column_id: 11513756
-          github_release_name: ${{ github.event.release.tag_name }}
-          github_token: ${{ secrets.ORGSCOPED_GITHUB_TOKEN }}
+          name: release-notes
+          path: release-notes.txt
+          retention-days: 1
+  terraform-provider-release:
+    name: 'Terraform Provider Release'
+    needs: [go-version, release-notes]
+    uses: hashicorp/ghaction-terraform-provider-release/.github/workflows/hashicorp.yml@v1
+    secrets:
+      hc-releases-aws-access-key-id: '${{ secrets.TF_PROVIDER_RELEASE_AWS_ACCESS_KEY_ID }}'
+      hc-releases-aws-secret-access-key: '${{ secrets.TF_PROVIDER_RELEASE_AWS_SECRET_ACCESS_KEY }}'
+      hc-releases-aws-role-arn: '${{ secrets.TF_PROVIDER_RELEASE_AWS_ROLE_ARN }}'
+      hc-releases-fastly-api-token: '${{ secrets.HASHI_FASTLY_PURGE_TOKEN }}'
+      hc-releases-github-token: '${{ secrets.HASHI_RELEASES_GITHUB_TOKEN }}'
+      hc-releases-terraform-registry-sync-token: '${{ secrets.TF_PROVIDER_RELEASE_TERRAFORM_REGISTRY_SYNC_TOKEN }}'
+      setup-signore-github-token: '${{ secrets.HASHI_SIGNORE_GITHUB_TOKEN }}'
+      signore-client-id: '${{ secrets.SIGNORE_CLIENT_ID }}'
+      signore-client-secret: '${{ secrets.SIGNORE_CLIENT_SECRET }}'
+    with:
+      goreleaser-release-args: --timeout 2h
```

Comment:
just to be on the safe side..in practice it took about 1h20min total from start to finish without publishing to the registry